sdk/python/kfp_tekton/compiler/_op_to_template.py

```diff
@@ -319,7 +319,32 @@ def _process_output_artifacts(outputs_dict: Dict[Text, Any],
         mounted_artifact_paths = []
         for artifact in outputs_dict['artifacts']:
             artifact_name = artifact_to_result_mapping.get(artifact['name'], artifact['name'])
-            if artifact['name'] in replaced_param_list:
+            if artifact['path'] == 'logs_to_S3':
+                copy_artifacts_step['script'] = copy_artifacts_step['script'] + \
+                    'mc cp /var/log/containers/$TASKRUN*$NAMESPACE*.log' + \
+                    ' storage/%s/artifacts/$PIPELINERUN/$PIPELINETASK/\n' % (bucket)
+                copy_artifacts_step['env'].append({
+                    'name': 'TASKRUN',
+                    'valueFrom': {
+                        'fieldRef': {
+                            'fieldPath': "metadata.labels['tekton.dev/taskRun']"
+                        }
+                    }
+                })
+                volume_mount_step_template.extend([{
+                    'name': 'varlog',
+                    'mountPath': '/var/log'
+                }, {
+                    'name': 'varlibdockercontainers',
+                    'mountPath': '/var/lib/docker/containers',
```

Comment:
Current I'm see this when running on IBM Cloud.
```
Added `storage` successfully.
mc: <ERROR> Unable to validate source /var/log/containers/parallel-pipeline-9876e-gcs-download-9b5qx-pod-7xckz_kubeflow_place-scripts-9610f8d694a83791f57386931f55d430775fa3fe3ac2cd4f48182f8e40f51b2d.log
```

This path only has the log symlink for Docker minikube. For IBM Cloud and other cloud provider, they might have different symlink for the log path. Take a look at the example from knative where it covers most of the symlink use case for all cloud providers.
https://github.com/knative/serving/blob/master/config/monitoring/logging/elasticsearch/200-fluentd.yaml#L122-L134

Also, mounting host path for every pipeline could create some security concern. What do you think about converting this part into a daemon set and just use this step to send the PIPELINERUN/PIPELINETASK information to the daemon set?

Reply:
Thanks @Tomcli , add varlibkubeletpods and varlogpods same as knative